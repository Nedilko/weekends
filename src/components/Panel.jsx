import PropTypes from 'prop-types'

Panel.propTypes = {
  children: PropTypes.node.isRequired,
  size: PropTypes.string,
}

function Panel({ children, size = 'sm' }) {
  return (
    <div
      className={`mx-auto max-w-${size} bg-white p-4 dark:bg-zinc-900 rounded-xl shadow-md`}
      onClick={(e) => e.stopPropagation()}
    >
      {children}
    </div>
  )
}

export default Panel
