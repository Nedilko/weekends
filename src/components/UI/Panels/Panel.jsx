import PropTypes from 'prop-types'

Panel.propTypes = {
  children: PropTypes.node.isRequired,
  size: PropTypes.string,
}

Panel.defaultProps = {
  size: 'sm',
}

function Panel({ children, size }) {
  return (
    <div
      className={`max-w-${size} mx-auto rounded-xl bg-white p-4 shadow-md dark:bg-zinc-900`}
      onClick={(e) => e.stopPropagation()}
    >
      {children}
    </div>
  )
}

export default Panel
