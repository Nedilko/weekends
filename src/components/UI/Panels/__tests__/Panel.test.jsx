import Panel from '@UI/Panels/Panel'
import { render, screen } from '@utils/test-utils'

describe('panel', () => {
  it('renders text', () => {
    const { container } = render(<Panel>some text</Panel>)
    expect(screen.getByText('some text')).toBeInTheDocument()
    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
