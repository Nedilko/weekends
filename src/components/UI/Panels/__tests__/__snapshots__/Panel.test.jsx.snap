// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should render content 1`] = `
<div>
  <div
    class="max-w-sm mx-auto rounded-xl bg-white p-4 shadow-md dark:bg-zinc-900"
  >
    some content
  </div>
</div>
`;

exports[`Panel should render with size 1`] = `
<div>
  <div
    class="max-w-md mx-auto rounded-xl bg-white p-4 shadow-md dark:bg-zinc-900"
  >
    some content
  </div>
</div>
`;
